amlip_cpp/src/cpp/types/GenericType.hpp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file GenericType.hpp
+ * This header file contains the declaration of a generic type that contains void* data.
+ */
+
+#ifndef AMLIP__SRC_CPP_TYPES_GENERICTYPE_HPP
+#define AMLIP__SRC_CPP_TYPES_GENERICTYPE_HPP
+
+#include <atomic>
+
+#include <types/InterfaceDataType.hpp>
+
+namespace eprosima {
+namespace fastcdr {
+class Cdr;
+} // namespace fastcdr
+} // namespace eprosima
+
+
+namespace eprosima {
+namespace amlip {
+namespace types {
+
+/*!
+ * @brief This class represents the structure GenericType, which allows to send any type of data as a void*.
+ * This is, data is streamed as an array of octets given its size in memory regardless of its original type/structure.
+ * @ingroup AMLIP
+ */
+class GenericType : public InterfaceDataType
+{
+public:
+
+    /*!
+     * @brief Default constructor with values.
+     */
+    GenericType(
+        void* data,
+        const uint32_t size);
+
+    /*!
+     * @brief Default constructor.
+     */
+    GenericType();
+
+    /*!
+     * @brief Default destructor.
+     */
+    ~GenericType();
+
+    /*!
+     * @brief Copy constructor.
+     * @param x Reference to the object GenericType that will be copied.
+     */
+    GenericType(
+            const GenericType& x);
+
+    /*!
+     * @brief Move constructor.
+     * @param x Reference to the object GenericType that will be copied.
+     */
+    GenericType(
+            GenericType&& x);
+
+    /*!
+     * @brief Copy assignment.
+     * @param x Reference to the object GenericType that will be copied.
+     */
+    GenericType& operator =(
+            const GenericType& x);
+
+    /*!
+     * @brief Move assignment.
+     * @param x Reference to the object GenericType that will be copied.
+     */
+    GenericType& operator =(
+            GenericType&& x);
+
+    /*!
+     * @brief Comparison operator.
+     * @param x GenericType object to compare.
```

Comment:
Comment explicitly that it compares that the memory space they refer is the same, and do not compare the actual data.